File: Challenge 10.py
'''
Make a game of rock, paper scissors against the
computer.
Algorithm
 Tell user to enter either rock,paper or scissors
 Get the response
 Generate a random number from 1 to 3 (1=rock,2=paper,
 3=scissors)
 Compare user selection and computer selection
 Display who wins.

Extension
Make sure the user enters a valid entry.
Add a loop structure to play several times and keep a running
score
Make an enumerated variable type to store choices. 
'''
import random, sys
print('ROCK, PAPER, SCISSORS')

wins = 0
losses = 0
ties = 0

while True:
    print(' %s Wins. %s Losses, %s Ties'%(wins, losses, ties))
    while True:
        print('Enter your move: (r)ock, (p)aper, (s)cissors or (q)uit.')
        player_move = input()
        if player_move == 'q':
            sys.exit()
        if player_move == 'r' or player_move == 'p' or player_move == 's':
            break
        print('Type one of r, p, s or q.')

    if player_move == 'r':
        print('ROCK versus...')
    elif player_move == 'p':
        print('PAPER versus...')
    elif player_move == 's':
        print('SCISSORS versus...')

    random_number = random.randint(1, 3)
    if random_number == 1:
        computer_move = 'r'
        print('ROCK')
    elif random_number == 2:
        computer_move = 'p'
        print('PAPER')
    elif random_number == 3:
        computer_move == 's'
        print('SCISSORS')

    if player_move == computer_move:
        print('It is a tie!')
        ties += 1
    elif player_move == 'r' and computer_move == 's':
        print('You win!')
        wins += 1
    elif player_move == 'p' and computer_move == 'r':
        print('You win!')
        wins += 1
    elif player_move == 's' and computer_move == 'p':
        print('You win!')
        wins += 1
    elif player_move == 'r' and computer_move == 'p':
        print('You lose!)')
        losses += 1
    elif player_move == 'p' and computer_move == 's':
        print('You lose!')
        losses += 1
    elif player_move == 's' and computer_move == 'r':
        print('You lose!')
        losses += 1

File: Challenge 10.java
import java.util.Random;
import java.util.Scanner;

class RockPaperScissors {
    static int wins = 0;
    static int losses = 0;
    static int ties = 0;

    public static String getUserInput() {
        System.out.print(">> ");
        Scanner sc_input = new Scanner(System.in);
        String input = sc_input.nextLine();
        return input;
    }

    public static void getResult(String player, String computer) {
        if (player.equals(computer)) {
            System.out.println("It's a tie.");
            ties += 1;
        } else if (player.toLowerCase().equals("r") && computer.equals("s")) {
            System.out.println("You win.");
            wins += 1;
        } else if (player.toLowerCase().equals("p") && computer.equals("r")) {
            System.out.println("You win.");
            wins += 1;
        } else if (player.toLowerCase().equals("s") && computer.equals("p")) {
            System.out.println("You win.");
            wins += 1;
        } else {
            System.out.println("You loose.");
            losses += 1;
        }
    }

    public static String getComputerMove() {
        String[] hand = { "r", "p", "s" };
        Random r = new Random();
        int random_index = r.nextInt(hand.length);
        return hand[random_index];
    }

    public static void printChoices(String usr_choice, String cmp_choice) {
        if (usr_choice.equals("s")) {
            System.out.println("SCISSORS VS... ");
        } else if (usr_choice.equals("p")) {
            System.out.println("PAPER VS... ");
        } else if (usr_choice.equals("r")) {
            System.out.println("ROCK VS... ");
        } else {
            System.out.println("Are you trying to cheat?");
        }
        if (cmp_choice.equals("r")) {
            System.out.println("ROCK");
        } else if (cmp_choice.equals("s")) {
            System.out.println("SCISSORS");
        } else if (cmp_choice.equals("p")) {
            System.out.println("PAPER");
        }
    }
    
    public static void main(String[] args) {
        System.out.println("Enter your move: (r)ock, (p)aper, (s)cissors or (q)uit.");
        boolean run = true;
        while (run == true) {
            System.out.println(wins + " Wins. " + losses + " Losses. " + ties + " Ties. ");
            String usr_choice = getUserInput().toLowerCase();
            String cmp_choice = getComputerMove();
            if (usr_choice.equals("q")) {
                run = false;
            }
            if (run == true) {
                printChoices(usr_choice, cmp_choice);
                getResult(usr_choice, cmp_choice);
                System.out.println();
            }
        }
    }
}
